src/modules/function/lines.rs

```diff
@@ -0,0 +1,77 @@
+use heraclitus_compiler::prelude::*;
+use crate::docs::module::DocumentationModule;
+use crate::modules::expression::expr::Expr;
+use crate::modules::types::{Type, Typed};
+use crate::translate::module::TranslateModule;
+use crate::utils::metadata::{ParserMetadata, TranslateMetadata};
+
+#[derive(Debug, Clone)]
+pub struct LinesInvocation {
+    path: Box<Option<Expr>>,
+}
+
+impl Typed for LinesInvocation {
+    fn get_type(&self) -> Type {
+        Type::Array(Box::new(Type::Text))
+    }
+}
+
+impl SyntaxModule<ParserMetadata> for LinesInvocation {
+    syntax_name!("Lines Invocation");
+
+    fn new() -> Self {
+        LinesInvocation {
+            path: Box::new(None)
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        token(meta, "lines")?;
+        token(meta, "(")?;
+        let tok = meta.get_current_token();
+        let mut path = Expr::new();
+        syntax(meta, &mut path)?;
+        token(meta, ")")?;
+        if path.get_type() != Type::Text {
+            return error!(meta, tok, "Expected string");
```

Comment:
The type is called `Text` (I was thinking some time ago if we should rename it to `String` but everyone voted to keep that characteristic in this language and stay with `Text`). Also let's give more information about the type of the expression back to the user.

```suggestion
			   let msg = format!("Expected value of type Text. Got {} instead.", path.get_type())
            return error!(meta, tok, msg);
```

Reply:
Changed to "Expected array value of type 'Text', got '...'" to match similar code in array.rs:
```
 ERROR  Expected array value of type 'Text', got 'Num'
at ./concat.ab:2:19

1| #!/usr/bin/env amber
2| for line in lines(42) {
3|     echo line
```